### PR TITLE
e2e: Plugin search results are as expected

### PR DESCRIPTION
**Pull request by lsl**
#### Changes proposed in this Pull Request

* Tests that searching for "ecommerce" results in at least one plugin search result with the title "WooCommerce"

#### Testing instructions

[Prereq setup](https://github.com/Automattic/wp-calypso/tree/trunk/test/e2e)

```
cd e2e/tests
yarn workspace @automattic/calypso-e2e build && yarn jest specs/plugins/plugins__search.ts
yarn workspace @automattic/calypso-e2e build && VIEWPORT_NAME=mobile yarn jest specs/plugins/plugins__search.ts
```

Related to https://github.com/Automattic/wp-calypso/issues/61094

### CODE REVIEW

**Comment by lsl** [comment]
This will be failing build due to being based on the pricing PR but the search query test seems to work as expected.

File: packages/calypso-e2e/src/lib/pages/plugins-page.ts
Comment (dpasque) [inline packages/calypso-e2e/src/lib/pages/plugins-page.ts:120]: Small nit, but I would stop this function here, and break the last line out into its own `validate` function!

It will probably help with future flexibility in case people want to search on this page, but don't want to validate against a specific finding.

**Comment by lsl** [comment]
Well... that rebase pulled in some unexpected code owners.

**Review by dpasque — APPROVED** [review]
Looking great! 🚀

**Review by dpasque — COMMENTED** [review]
Oh whoops, it looks like the mobile version of this test is now failing! Sorry, saw that after doing the code review and approving.

It looks like there are mobile and desktop versions of the search input, and that the desktop one just gets hidden.

You can probably get around that using the [Playwright :visible psuedo-selector](https://playwright.dev/docs/selectors#selecting-visible-elements)! (to make sure it's always getting the visible search bar).

Once that's done you should be good to merge!

**Comment by lsl** [comment]
> You can probably get around that using the [Playwright :visible psuedo-selector](https://playwright.dev/docs/selectors#selecting-visible-elements)! (to make sure it's always getting the visible search bar). 

Thanks!

**Pull request merged**
